src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -439,6 +439,23 @@ private static bool ShouldUseInlineArray(BoundCollectionExpressionBase node, CSh
             Debug.Assert(_compilation.Assembly.RuntimeSupportsInlineArrayTypes);
 
             int arrayLength = elements.Length;
+            if (arrayLength == 1
+                && _factory.WellKnownMember(asReadOnlySpan
+                    ? WellKnownMember.System_ReadOnlySpan_T__ctor_ref_readonly_T
+                    : WellKnownMember.System_Span_T__ctor_ref_T, isOptional: true) is MethodSymbol spanRefConstructor)
+            {
+                // Special case: no need to create an InlineArray1 type. Just use a temp of the element type.
+                var spanType = _factory
+                    .WellKnownType(asReadOnlySpan ? WellKnownType.System_ReadOnlySpan_T : WellKnownType.System_Span_T)
+                    .Construct([elementType]);
+                var constructor = spanRefConstructor.AsMember(spanType);
+                var element = VisitExpression((BoundExpression)elements[0]);
+                var temp = _factory.StoreToTemp(element, out var assignment, isKnownToReferToTempIfReferenceType: true);
```

Comment:
>isKnownToReferToTempIfReferenceType [](http://example.com/codeflow?start=77&length=35)

1. In general, when not sure, the safest value for this parameter is `false`. It enables some optimizations and you don't want to apply those when they are not appropriate because that can lead to an incorrect result, i.e. bugs.
2. This value is meaningful only for ref locals. I would avoid setting it to true for non-ref locals.
3. The meaning of this flag - this local refers to something (it is a ref), and if the type of that something is a reference type, there is a guarantee that that something is a temporary (not a user local, not a parameter, not a field, etc.) So, it is about what we are referring to with this local. Obviously this temp is a temp, but this value is not about that.

So, I recommend, cleaning up this code and any other "expression collections" code with respect to values used for this parameter. In order to avoid spreading questionable code through copy/paste and eventually hitting a scenario that can actually break because we haven't given much thought about the value.